dbus: don't unlink monitors list if it is NULL

Avoid a segfault in bus_connection_disconnected when the monitors
list was never allocated.

Upstream-Status: Inappropriate [OpenXT]

--- a/bus/connection.c	2026-09-10 11:20:01.607744918 -0400
+++ b/bus/connection.c	2026-09-10 11:20:01.607744918 -0400
@@ -306,7 +306,8 @@
       if (mm != NULL)
         bus_matchmaker_disconnected (mm, connection);
 
-      _dbus_list_remove_link (&d->connections->monitors, d->link_in_monitors);
+      if (d->connections->monitors != NULL)
+        _dbus_list_remove_link (&d->connections->monitors, d->link_in_monitors);
       d->link_in_monitors = NULL;
     }
 
